File: Serene/Serene.Script/Imports/ServiceContracts/Administration.RolePermissionService.cs
namespace Serene.Administration
{
    using jQueryApi;
    using Serenity;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [Imported, PreserveMemberCase]
    public partial class RolePermissionService
    {
        [InlineConstant] public const string BaseUrl = "Administration/RolePermission";
    
        [InlineCode("Q.serviceRequest('Administration/RolePermission/Update', {request}, {onSuccess}, {options})")]
        public static jQueryXmlHttpRequest Update(RolePermissionUpdateRequest request, Action<SaveResponse> onSuccess, ServiceCallOptions options = null)
        {
            return null;
        }
    
        [InlineCode("Q.serviceRequest('Administration/RolePermission/List', {request}, {onSuccess}, {options})")]
        public static jQueryXmlHttpRequest List(RolePermissionListRequest request, Action<RolePermissionListResponse> onSuccess, ServiceCallOptions options = null)
        {
            return null;
        }
    
        [Imported, PreserveMemberCase]
        public static class Methods
        {
            [InlineConstant] public const string Update = "Administration/RolePermission/Update";
            [InlineConstant] public const string List = "Administration/RolePermission/List";
        }
    }
    
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Administration.RolePermissionService.ts
namespace Serene.Administration {
    export namespace RolePermissionService {
        export const baseUrl = 'Administration/RolePermission';

        export declare function Update(request: RolePermissionUpdateRequest, onSuccess?: (response: Serenity.SaveResponse) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;
        export declare function List(request: RolePermissionListRequest, onSuccess?: (response: RolePermissionListResponse) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;

        export namespace Methods {
            export declare const Update: 'Administration/RolePermission/Update';
            export declare const List: 'Administration/RolePermission/List';
        }

        ['Update', 'List'].forEach(x => {
            (<any>RolePermissionService)[x] = function (r, s, o) { return Q.serviceRequest(baseUrl + '/' + x, r, s, o); };
            (<any>Methods)[x] = baseUrl + '/' + x;
        });
    }
}

